atsign-foundation/at_server
# feat: Enforce superset access check for approving apps

**- What I did**
- Implement a verification process ensuring that only client possessing the super-set privileges or clients authorized for the specified namespaces in the enrollment request can approve it.
- Update the minimum SDK version in pubspec.yaml to "3.0.0" to support dart 3 new features.

**- How I did it**
- Refactor the "isAuthorised" method in the "abstract_verb_handler.dart" file to enforce the validation described above.
   - Remove "_isAuthorised" method and move/refactor the logic to "isAuthorised" method.
   - Introduce "_checkForNamespaceAuthorization" method, which will run the namespace authorisation check with the enrolled namespace.
   - Implement a method called "checkEnrollmentNamespaceAccess" to verify authorized namespace access according to the verb used. This method is overrides in enroll_verb_handler.dart,  to validate access of namespaces in the enrollment requests against the authorized namespace access.

**- How to verify it**
- Added the following unit tests:
   - A test to verify that the authorization check returns false when the client is not authorized for the requested namespace
   - A test to verify that the authorization check returns true when the client is authorized to the namespace
   - A test to verify that the authorization check returns true when the client is authorized for manage namespace
   - A test to verify that the authorization check returns true when the client is authorized to * namespaces
   - A test to verify that the authorization check returns true when the client is PKAM authentication and enrollment id is null 
 - Added the following functional tests:
   - A test to assert that an authenticated connection without namespace authorization cannot approve-deny requests for that namespace
   - A test to assert that an authenticated connection without namespace authorization cannot revoke request for that namespace

**- Description for the changelog**
- Enforce superset access check for approving apps
